openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -434,6 +461,74 @@ def create_library_block_child(parent_usage_key, block_type, definition_id):
     return metadata
 
 
+def get_library_block_static_asset_files(usage_key):
+    """
+    Given an XBlock in a content library, list all the static asset files
+    associated with that XBlock.
+
+    Returns a list of LibraryXBlockStaticFile objects.
+    """
+    def_key, lib_bundle = _lookup_usage_key(usage_key)
+    result = [
+        LibraryXBlockStaticFile(path=f.path, url=f.url, size=f.size)
+        for f in lib_bundle.get_static_files_for_definition(def_key)
+    ]
+    result.sort(key=lambda f: f.path)
+    return result
+
+
+def add_library_block_static_asset_file(usage_key, file_name, file_content):
+    """
+    Upload a static asset file into the library, to be associated with the
+    specified XBlock. Will silently overwrite an existing file of the same name.
+
+    file_name should be a name like "doc.pdf". It may optionally contain slashes
+        like 'en/doc.pdf'
+    file_content should be a binary string.
+
+    Returns a LibraryXBlockStaticFile object.
+
+    Example:
+        video_block = UsageKey.from_string("lb:VideoTeam:python-intro:video:1")
+        add_library_block_static_asset_file(video_block, "subtitles-en.srt", subtitles.encode('utf-8'))
+    """
+    assert isinstance(file_content, six.binary_type)
+    def_key, lib_bundle = _lookup_usage_key(usage_key)
+    if file_name != file_name.strip().strip('/'):
+        raise InvalidNameError("file name cannot start/end with / or whitespace.")
+    if '//' in file_name or '..' in file_name:
+        raise InvalidNameError("Invalid sequence (// or ..) in filename.")
+    file_path = lib_bundle.get_static_prefix_for_definition(def_key) + file_name
+    # Write the new static file into the library bundle's draft
+    draft = get_or_create_bundle_draft(def_key.bundle_uuid, DRAFT_NAME)
+    write_draft_file(draft.uuid, file_path, file_content)
+    # Clear the bundle cache so everyone sees the new file immediately:
+    lib_bundle.cache.clear()
+    file_metadata = blockstore_cache.get_bundle_file_metadata_with_cache(
+        bundle_uuid=def_key.bundle_uuid, path=file_path, draft_name=DRAFT_NAME,
+    )
+    return LibraryXBlockStaticFile(path=file_metadata.path, url=file_metadata.url, size=file_metadata.size)
+
+
+def delete_library_block_static_asset_file(usage_key, file_name):
+    """
+    Delete a static asset file from the library.
+
+    Example:
+        video_block = UsageKey.from_string("lb:VideoTeam:python-intro:video:1")
+        delete_library_block_static_asset_file(video_block, "subtitles-en.srt")
+    """
+    def_key, lib_bundle = _lookup_usage_key(usage_key)
+    if '..' in file_name:
+        raise InvalidNameError("Invalid .. in file name.")
```

Comment:
We probably don't, because I believe Blockstore does the same validation. I'm just being a bit extra cautious. We don't want to allow reads or writes to any content outside of the `..../static/` "folder"